R/analyse_pseudo_bulk.R

```diff
@@ -0,0 +1,373 @@
+  }
+  
+  if(split_by == "Louvain"){
+    resolution_ <- k_variable[[1]]
+    subcluster_ref <- k_variable[[2]]   # subcluster variable depending on entry in metadata
+    subcluster_comp <- k_variable[[3]]  # subcluster variable depending on entry in metadata
+
+    if(!is.numeric(resolution_)){
+      stop('Error: resolution must be a number')
+    }
+
+    result <- split_clustering(seurat_object, group_by,resolution_, 1, subcluster_ref,subcluster_comp)
+    return(result)
+  }
+  
+  if(split_by == "Louvain_multilevel"){
+    resolution_ <- k_variable[[1]]
+    subcluster_ref <- k_variable[[2]]   # subcluster variable depending on entry in metadata
+    subcluster_comp <- k_variable[[3]]  # subcluster variable depending on entry in metadata
+   
+    if(!is.numeric(resolution_)){
+      stop('Error: resolution must be a number')
+    }
+   
+    result <- split_clustering(seurat_object, group_by,resolution_, 2, subcluster_ref,subcluster_comp)
+    return(result)
+  }
+  
+  if(split_by =="SLM"){
+    resolution_ <- k_variable[[1]]
+    subcluster_ref <- k_variable[[2]]    # subcluster variable depending on entry in metadata
+    subcluster_comp <- k_variable[[3]]   # subcluster variable depending on entry in metadata
+
+
+    if(!is.numeric(resolution_)){
+      stop('Error: resolution must be a number')
+    }
+    
+    result <- split_clustering(seurat_object, group_by,resolution_, 3, subcluster_ref,subcluster_comp)
+    return(result)
+  }
+  
+  if(split_by =="Leiden"){
+    resolution_ <- k_variable[[1]]
+    subcluster_ref <- k_variable[[2]]    # subcluster variable depending on entry in metadata
+    subcluster_comp <- k_variable[[3]]   # subcluster variable depending on entry in metadata
+  
+    if(!is.numeric(resolution_)){
+      stop('Error: resolution must be a number')
+    }
+
+    result <- split_clustering(seurat_object, group_by,resolution_, 4, subcluster_ref,subcluster_comp)
+    return(result)
+  }
+})
+
+
+
+#' generate_pseudo_bulk_data using SingleCellExperiment
+#' @inherit generate_pseudo_bulk_data 
+#' @param group_by      entry in metadata table, based on these cluster annotation pseudo bulk is performed
+#' @param split_by      variable -> split by a variable within the metadata; k must be a string
+#'                      random -> splits based on a random number; k must be a number
+#'                      subclustering [resolution, cluster_1, cluster_2]
+#' @param k_variable    variable dependent on the split_by
+#'
+#' @returns             returns pseudo bulk generated data
+setMethod("generate_pseudo_bulk_data", c("object" = "SingleCellExperiment"), function(object, 
+                                                                                      group_by, 
+                                                                                      split_by, 
+                                                                                      k_variable){
+
+  if (!is.character(split_by)) {
```

Comment:
What happens if the user has a typo in `split_by`? I guess, the function then returns nothing.
Ideally, you should call `stop` in this case